awscli/customizations/cloudformation/deployer.py

```diff
@@ -106,6 +108,26 @@ def create_changeset(self, stack_name, cfn_template,
             'Capabilities': capabilities,
             'Description': description,
         }
+
+        # CloudFormation requires template bodies to be less than 51200 bytes
+        # if they are deployed from a file. It supports much larger templates
+        # if the templates exist in s3. If this is a 'big' template, throw it
+        # into s3 first.
+        cfn_template_length = len(cfn_template.encode('utf-8'))
+        if cfn_template_length > 51200:
+            if s3_uploader:
+                with mktempfile() as temporary_file:
+                    temporary_file.write(kwargs.pop('TemplateBody'))
+                    temporary_file.flush()
+                    url = s3_uploader.upload_with_dedup(
+                            temporary_file.name, "template")
+                    kwargs['TemplateURL'] = s3_uploader.to_path_style_s3_url(url.split('/')[-1])
+            else:
+                LOG.debug("Template size is too large and S3 bucket not given")
+                raise ValueError(("A template with a size greater than 51200 "
```

Comment:
For example usage, checkout https://github.com/aws/aws-cli/blob/50b4ccfb85acb714445cbb4a6fb0582ade8af2b0/awscli/customizations/cloudformation/artifact_exporter.py#L141